app/src/main/java/com/kamron/pogoiv/ScreenGrabber.java

```diff
@@ -144,4 +114,55 @@ private Bitmap getBitmap(ByteBuffer buffer, int pixelStride, int rowPadding) {
 
         return bmp;
     }
+
+    public @Nullable @ColorInt int[] grabPixels(Point[] points) {
+        Image image = null;
+        try {
+            //Note: imageReader shouldn't be null, but apparently sometimes is.
+            //Let's allow this to still happen.
+            image = imageReader.acquireLatestImage();
+        } catch (Exception exception) {
+            Timber.e("Error thrown in grabPixels() - acquireLatestImage()");
+            Timber.e(exception);
+        }
+
+        if (image == null) {
+            return null;
+        }
+
+        Rect imageBounds = new Rect(0, 0, image.getWidth(), image.getHeight());
+        @ColorInt int[] pixels = new int[points.length];
+
+        Image.Plane imgPlane = image.getPlanes()[0];
+        int pixelStride = imgPlane.getPixelStride();
+        int rowStride = imgPlane.getRowStride();
+        ByteBuffer buffer = imgPlane.getBuffer();
+
+        for (int i = 0; i < points.length; i++) {
+            Point p = points[i];
+            if (imageBounds.contains(p.x, p.y)) {
+                pixels[i] = getPixel(buffer, p, pixelStride, rowStride);
+            } else {
+                pixels = null;
+                //Jump to resource cleanup code.
+                break;
+            }
+        }
+
+        //Resource cleanup
+        image.close();
+
+        return pixels;
+    }
+
+    //Inspired by http://stackoverflow.com/a/27655022/53974.
+    private static @ColorInt int getPixel(ByteBuffer buffer, Point pos, int pixelStride, int rowStride) {
+        int offset = pos.y * rowStride + pos.x * pixelStride;
+        int pixel = 0;
+        pixel |= (buffer.get(offset) & 0xff) << 16;     // R
+        pixel |= (buffer.get(offset + 1) & 0xff) << 8;  // G
+        pixel |= (buffer.get(offset + 2) & 0xff);       // B
+        pixel |= (buffer.get(offset + 3) & 0xff) << 24; // A
```

Comment:
Thanks for `Color.argb()`.
But note the image reader uses the `PixelFormat.RGBA_8888` format, not a non-existing `PixelFormat.ARGB_8888`—which is one reason why this code actually reorders the bytes.
Viceversa, the bitmaps we create use `Bitmap.Config.ARGB_8888` instead of a non-existing `Bitmap.Config.RGBA_8888`.

(Not that I realized either when I copy-pasted this, otherwise I'd have called this out in a comment).